src/Umbraco.Web/Mvc/RenderRouteHandler.cs

```diff
@@ -193,12 +193,18 @@ internal static IHttpHandler HandlePostedValues(RequestContext requestContext, P
                         surfaceRoute = surfaceRoutes.FirstOrDefault(x =>
                             x.Defaults["action"] != null &&
                             x.Defaults["action"].ToString().InvariantEquals(postedInfo.ActionName));
+
+                        if (surfaceRoute == null)
+                        {
+                            // we found n possibilities, but no exact matches in the Route Table.
+                            // is one of these a generic SurfaceController route?
+                            surfaceRoute = surfaceRoutes.Where(x => x.DataTokens["umbraco"].ToString().InvariantEquals("surface")).FirstOrDefault();
```

Comment:
Since we are only ever looking for surface controllers here, this logic should just be part of the initial linq query to resolve `surfaceRoutes` but this will also requre an existing key check like we are doing with `x.DataTokens.ContainsKey`

Reply:
@bythewiseman do you have some time to review this by chance?

Reply:
Do you want me to put that line of code in my forked code and test my scenario? Sure. It won't be until Sunday though if that is okay.

Reply:
Yes please, any changes you make to your fork on that branch are automatically applied to this PR